src/ast/ast.c

```diff
@@ -72,6 +72,22 @@ static void _AST_LimitResults(AST *ast, const cypher_astnode_t *root_clause,
 	}
 }
 
+static void _extract_params(const cypher_astnode_t *statement) {
+	rax *params = raxNew();
```

Comment:
I don't think it makes sense to allocate `params` unless the query actually has params. This will require some changes to NULL handling in `QueryCtx_GetParams` and its callers.

Reply:
changed to lazy initialization. tell me what you think